### PR TITLE
Make lightgbm-lib uploadable to remote repository

### PR DESCRIPTION
**Pull request by shenggwang**
With below dilemma:
On one hand, add <scope>provided</scope> makes our project stop to find it on the remote repository (due to not being a transitive dependency) but excludes lightgbm-lib.jar from our uber jar.
On the other hand, remove <scope>provided</scope> includes lightgbm-lib.jar to our uber jar but makes our project try to find it on the remote repository, which doesn’t exist.

This commit removes <scope>provided</scope>, and tries to deploy lightgbm-lib to the remote repository in order to let our project to find it on the remote repository.

------

The experimental tag doesn't test it, so there is no test that can 100% confirm this will work.
However, from previous deploy error as described on this [PR](https://github.com/feedzai/feedzai-openml-java/pull/46), it was due to missing javadoc.jar and source.jar, which in this commit, they are created.

Steps confirmed on the local machine:
1-maven install, confirm that on lightgbm-lib exists 3 files, jar, javadoc and source
2-maven install, confirm that openml-lightgbm uber jar contains ligthgbm-lib.
2-maven deploy has an unauthorized error

### CODE REVIEW

File: openml-lightgbm/lightgbm-builder/javadoc/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Ligthgbm Lib Javadoc
+This is a dummy javadoc.
```
Comment (AlbertoEAF) [inline openml-lightgbm/lightgbm-builder/javadoc/README.md:2]: Do we need this? If so move on :)

**Review by AlbertoEAF — COMMENTED** [review]
If it works then let's proceed :)

File: openml-lightgbm/lightgbm-builder/javadoc/README.md (hunk 1 above)
Comment (shenggwang) [reply openml-lightgbm/lightgbm-builder/javadoc/README.md:2]: Yes. The previous [PR](https://github.com/feedzai/feedzai-openml-java/pull/46) was made due to javadoc and source error. See its [description](https://github.com/feedzai/feedzai-openml-java/pull/46)

**Comment by JPDSousa** [comment]
@shenggwang have you tried this in an experimental?

**Review by JPDSousa — APPROVED** [review]

**Comment by shenggwang** [comment]
> @shenggwang have you tried this in an experimental?

Nop, because experimental won't try to deploy lightgbm-lib. therefore it won't conclude anything.
However, from previous deploy error as described on this [PR](https://github.com/feedzai/feedzai-openml-java/pull/46), it was due to missing javadoc.jar and source.jar, which in this commit, they are created.

**Pull request merged**